packages/runtime/src/next/client.ts

```diff
@@ -219,6 +216,10 @@ export class Makeswift {
     return getMakeswiftSiteVersion(previewData) ?? MakeswiftSiteVersion.Live
   }
 
+  static getPreviewMode(previewData: PreviewData): boolean {
+    return getMakeswiftSiteVersion(previewData) === MakeswiftSiteVersion.Working
+  }
```

Comment:
Added a method for deriving `previewMode` from `previewData`, see the Pages router example.